deployment/ccip/changeset/cs_active_candidate.go

```diff
@@ -63,28 +94,31 @@ func PromoteAllCandidatesChangeset(
 	}, nil
 }
 
-// SetCandidateExecPluginProposal calls setCandidate on the CCIPHome for setting up OCR3 exec Plugin config for the new chain.
-// TODO: make it conform to the ChangeSet interface.
+// SetCandidatePluginChangeset calls setCandidate on the CCIPHome for setting up OCR3 exec Plugin config for the new chain.
 func SetCandidatePluginChangeset(
-	state CCIPOnChainState,
 	e deployment.Environment,
-	nodes deployment.Nodes,
-	ocrSecrets deployment.OCRSecrets,
-	homeChainSel, feedChainSel, newChainSel uint64,
-	tokenConfig TokenConfig,
-	pluginType cctypes.PluginType,
+	cfg AddDonAndSetCandidateChangesetConfig,
```

Comment:
I think we want to parameterize the whole OCR config, similar to the initial add chain changeset. Very likely it'll vary by chain. Also ditto re passing nodeIDs and doing state based validation

Reply:
Makes sense, will switch up to do that